### ISSUE TITLE
maxDuration does not work for remix + vite

### ISSUE THREAD

**Issue by lifeiscontent** [vercel/remix#96]
following the docs using the `export const config = { ... }` as well as the vercel.json fails to deploy for me

https://vercel.com/docs/functions/configuring-functions/duration

**Comment by lifeiscontent** [comment]
@TooTallNate any ideas here? we're having a ton of traffic on our site today and would really like to increase this 😅

**Comment by TooTallNate** [comment]
Can you elaborate on "fails to deploy"? What are you seeing? Can you share a reproduction, or a deployment URL?

**Comment by lifeiscontent** [comment]
@TooTallNate yep, here you go: https://vercel.com/sort-xyz/sorthub/BPFNXzZGm6jrgnnNFLt6FKf2NbUC

**Comment by TooTallNate** [comment]
Ok, so ya, you don't configure `maxDuration` for Remix routes using `vercel.json`. Do it the `export const config = { ... }` way like described in the [Remix on Vercel](https://vercel.com/docs/frameworks/remix#serverless-function-configuration) docs page.

**Comment by lifeiscontent** [comment]
@TooTallNate I tried that as well, and the build wouldn't even compile, complained about something to do with some files in node modules already existing. let me revert to that commit and send over a URL

**Comment by lifeiscontent** [comment]
@TooTallNate here you go: https://vercel.com/sort-xyz/sorthub/3r7gRrzaifeqz7SDvJzWUpVhVmVw

**Comment by TooTallNate** [comment]
This error seems unrelated to the `export const config` part, but did you see this issue which looks similar? https://github.com/remix-run/remix/issues/8813

**Comment by lifeiscontent** [comment]
![Screenshot 2024-03-21 at 1 50 18 PM](https://github.com/vercel/remix/assets/180963/fe25bb2b-dde6-4a2e-8efe-3d465d1a0339)
this is the entire diff, removing it fixes the build

**Comment by lifeiscontent** [comment]
@TooTallNate ^

**Comment by lifeiscontent** [comment]
interestingly enough, adding the config to `root.tsx` does not break the build

**Comment by TooTallNate** [comment]
That is strange. I will relay that to the Remix team. If you could give a standalone reproduction that would help as well.

**Comment by lifeiscontent** [comment]
I'll see what I can do, thanks!

**Comment by TooTallNate** [comment]
Ya that would be helpful. I wasn't able to reproduce using the `vc init remix` template, and changing the config of both of those routes in the template to have `export const config = { maxDuration: 30 }` worked for me.
